src/renderers/WebGLRenderer.js

```diff
@@ -1821,6 +1821,10 @@ THREE.WebGLRenderer = function ( parameters ) {
 
 			return a.object.renderOrder - b.object.renderOrder;
 
+		} else if ( !!a.material.program && !!b.material.program && a.material.program.id !== b.material.program.id ) {
```

Comment:
Yep yep! buffer/texture uplod sounds good to me. I think I'm almost done with the "design". I have many things broken in local at the moment. Will push stuff tonight. Even if a lot of things are broken (will keep working builds though).